.github/workflows/ci.yml

```diff
@@ -425,6 +425,9 @@ jobs:
     # Pin glibc to 2.31 version for better compatibility.
     container: debian:bullseye
     steps:
+      - name: Install Git
+        run: apt-get update && apt-get install -y git
```

Comment:
@SleepySquash I don't understand why we install Git here, if it's already should have been installed, because the action `checkout` below clearly calls and uses it.

Reply:
@tyranron, `checkout` step doesn't use Git - it fallbacks to simple GET requests, AFAIK, when there's not `git` executable available: https://github.com/actions/checkout/blob/692973e3d937129bcbf40652eb9f2f61becf3332/src/git-source-provider.ts#L77

Without Git the `PubspecBuilder` simply won't work. This isn't critical, yet without the builder and the variables it defines the updates won't be delivered properly for Linux users on `main` branch (`v*` tags still will receive those updates, as we don't rely on `PubspecBuilder` there, only the tag).

```
[PubspecBuilder] Unable to properly generate `pubspec.g.dart` summary: `git` executable exited with code 128, 
stdout: 
stderr: fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
```

Reply:
@SleepySquash ah so... can we make it as a hard error in a separate PR? Because it was quite misleading that the CI job succeeded, however the build is incorrect.